standalone-metastore/metastore-server/src/main/sql/derby/hive-schema-4.1.0.derby.sql

```diff
@@ -171,7 +171,7 @@ CREATE TABLE "APP"."NOTIFICATION_LOG" (
     "EVENT_ID" BIGINT NOT NULL,
     "EVENT_TIME" INTEGER NOT NULL,
     "EVENT_TYPE" VARCHAR(32) NOT NULL,
-    "MESSAGE" CLOB,
+    "MESSAGE" VARCHAR(32672),
```

Comment:
Why are we changing CLOB to VARCHAR here? 

IIRC CLOB value can be up to 2,147,483,647 characters long. 
Changing to VARCHAR will limit the message to  32,672 characters .

Reply:
This datatype change is for derby. Derby never supported CLOB to CLOB comparisons, however datanucleus 5.x had some workarounds to make CLOB to CLOB comparisons work, whick are not present as part of datanucleus 6.x . These fields have been changed to VARCHAR as none of them seem to store values of such big size.

Reply:
Agreed.

Reply:
there is a function for that. I don't think we should be doing that change
````
  public String toVarChar(String column) {
    switch (dbType) {
      case DERBY:
        return String.format("CAST(%s AS VARCHAR(4000))", column);
      case ORACLE:
        return String.format("to_char(%s)", column);
      default:
        return column;
    }
  }
````